enterprise/ssl-impl/src/test/java/io/crate/protocols/postgres/SslReqHandlerIntegrationTest.java

```diff
@@ -56,16 +85,32 @@ protected Settings nodeSettings(int nodeOrdinal) {
             .put(super.nodeSettings(nodeOrdinal))
             .put(SslConfigSettings.SSL_PSQL_ENABLED.getKey(), true)
             .put(SslConfigSettings.SSL_KEYSTORE_FILEPATH.getKey(), keyStoreFile.getAbsolutePath())
-            .put(SslConfigSettings.SSL_KEYSTORE_PASSWORD.getKey(), "keystorePassword")
-            .put(SslConfigSettings.SSL_KEYSTORE_KEY_PASSWORD.getKey(), "serverKeyPassword")
-            .put(SslConfigSettings.SSL_TRUSTSTORE_FILEPATH.getKey(), trustStoreFile.getAbsolutePath())
-            .put(SslConfigSettings.SSL_TRUSTSTORE_PASSWORD.getKey(), "truststorePassword")
             .build();
     }
 
     @Test
-    public void testCheckEncryptedConnection() throws Throwable {
-        SQLResponse response = execute("select name from sys.nodes");
-        assertEquals(1, response.rowCount());
+    public void testReloadSslContextOnKeyStoreChange() throws Throwable {
+        try {
+            execute("select name from sys.nodes");
+            fail("Unknown certificate in the certificate chain");
+        } catch (Exception ignored) {
+        }
+
+        updateKeyStore(
+            keyStoreFile,
+            keyStore -> keyStore.setKeyEntry(
+                "key",
+                trustedCert.key(),
+                EMPTY_PASS,
+                new Certificate[]{trustedCert.cert()}));
+
+        assertBusy(() -> {
+            try {
+                SQLResponse response = execute("select name from sys.nodes");
+                assertEquals(1, response.rowCount());
+            } catch (Exception e) {
+                fail();
+            }
+        }, 30, TimeUnit.SECONDS);
```

Comment:
Maybe we should make the interval configurable so that we can lower it in the tests and don't require as much time here.